nengo_gui/components/component.py

```diff
@@ -31,9 +35,9 @@ class Template(object):
     def __init__(self, *args, **kwargs):
         self.args = args
         self.kwargs = kwargs
-    def create(self, vizsim):
-        uid = vizsim.viz.get_uid(self)
-        c = self.cls(vizsim, vizsim.viz.config[self], uid,
+    def create(self, sim):
+        uid = '_uid_%d' % id(self)
+        c = self.cls(sim, sim.config[self], uid,
```

Comment:
How does this work? I can't find the `cls` attribute defined anywhere and it doesn't seem to be a default attribute.